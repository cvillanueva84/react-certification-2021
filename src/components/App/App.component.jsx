import React, { useState, useEffect } from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { lightTheme, darkTheme } from '../../utils/themes';

import AuthProvider from '../../providers/Auth';
import HomePage from '../../pages/Home';
import LoginPage from '../../pages/Login';
import NotFound from '../../pages/NotFound';
import SecretPage from '../../pages/Secret';
import Private from '../Private';
import Layout from '../Layout';

function App() {
  const [theme, setTheme] = useState(true); // true is light and false is dark
  const themeToggler = () => setTheme(!theme);
  useEffect(() => {
    document.title = 'Wizeline TV';
  }, []);
  return (
    <BrowserRouter>
      <AuthProvider>
        <ThemeProvider theme={theme ? lightTheme : darkTheme}>
          <Layout themeToggler={themeToggler}>
            <Switch>
              <Route exact path="/">
                <HomePage />
              </Route>
              <Route exact path="/login">
                <LoginPage />
              </Route>
              <Private exact path="/secret">
                <SecretPage />
              </Private>
              <Route path="*">
                <NotFound />
              </Route>
            </Switch>
          </Layout>
        </ThemeProvider>
      </AuthProvider>
    </BrowserRouter>
  );
}

export default App;
