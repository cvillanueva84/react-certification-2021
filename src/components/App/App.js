/* eslint-disable react/jsx-filename-extension */
import React from 'react';

import { GlobalStyles } from '../../GlobalStyles.styles';
import HomePage from '../../pages/Home/Home.page';

function App() {
  return (
    <div className="App">
      <GlobalStyles />
      <HomePage />
    </div>
  );
}

export default App;
