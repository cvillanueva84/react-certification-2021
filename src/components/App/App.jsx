import React from 'react';

import HomePage from '../../pages/Home/Home.page';
import data from '../Files/youtube-videos-mock.json';

function App() {
  return <HomePage data={data} />;
}

export default App;
