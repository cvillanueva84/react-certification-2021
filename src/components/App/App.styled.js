import styled from 'styled-components';
import breakpoint from '../../common/breakpoint';

const Header = styled.nav`
  background: #de382a;
  color: #fff;
  width: 100%;
  display: flex;
  box-sizing: border-box;
  flex-shrink: 0;
  flex-direction: column;
  min-height: 60px;
  padding-left: 24px;
  padding-right: 24px;
  .navbar {
    display: flex;
    height: 100%;
    align-items: center;
    justify-content: center;
    .spacer {
      flex: 1;
      @media only screen and (${breakpoint.device.xs}) {
        display: none;
      }
      @media only screen and (${breakpoint.device.sm}) {
        display: flex;
        flex: 1;
      }
      @media only screen and (${breakpoint.device.lg}) {
        display: flex;
        flex: 1;
      }
    }
  }
  input {
    margin-left: 1em;
    border-radius: 5px;
    border: none;
    padding: 8px 15px;
    cursor: text;
    display: inline-flex;
    position: relative;
    font-size: 1rem;
    box-sizing: border-box;
    align-items: center;
    font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
    font-weight: 400;
    line-height: 1.1876em;
    letter-spacing: 0.00938em;
    background: #bf2e21;
    color: white;
    padding-left: 40px;
    @media only screen and (${breakpoint.device.xs}) {
      width: 100%;
    }
  }
  input:focus {
    outline: 1px solid #fff;
  }
  .navChild {
    display: flex;
    align-items: center;
    justify-content: center;
    position: relative;
    @media only screen and (${breakpoint.device.xs}) {
      width: 100%;
    }
    @media only screen and (${breakpoint.device.sm}) {
      width: auto;
    }
    @media only screen and (${breakpoint.device.lg}) {
      width: auto;
    }
  }

  .navChild-right {
    @media only screen and (${breakpoint.device.xs}) {
      display: none;
    }
    @media only screen and (${breakpoint.device.sm}) {
      display: flex;
    }
    @media only screen and (${breakpoint.device.lg}) {
      display: flex;
    }
  }
`;

export { Header };
