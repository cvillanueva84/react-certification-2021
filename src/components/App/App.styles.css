.dark-mode {
  background-color: #333;
}
