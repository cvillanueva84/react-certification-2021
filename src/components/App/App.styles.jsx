import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  html {
    font-size: 1.125rem;
    line-height: 1.6;
    font-weight: 400;
    font-family: sans-serif;
    box-sizing: border-box;
    scroll-behavior: smooth;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  }
  div{
    color: ${(props) => props.theme.a};
  }
  .form-group {  background-color: ${(props) => props.theme.body};}
  #cardBody{
    box-shadow: ${(props) => props.theme.a} 1px 1px 0, ${(props) =>
  props.theme.a} 2px 2px 0, ${(props) => props.theme.a} 3px 3px 0, ${(props) =>
  props.theme.a} 4px 4px 0,
      ${(props) => props.theme.a} 5px 5px 0, ${(props) => props.theme.a} 6px 6px 0, ${(
  props
) => props.theme.a} 7px 7px 0, ${(props) => props.theme.a} 8px 8px 0;
    :hover {
      background-color: ${(props) => props.theme.cardBodyHover};
      box-shadow: red 1px 1px 0, red 2px 2px 0, red 3px 3px 0, red 4px 4px 0, red 5px 5px 0,
        red 6px 6px 0, red 7px 7px 0, red 8px 8px 0;
    }
  }
  #relatedVideo{
    :hover {
      background-color: ${(props) => props.theme.cardBodyHover};
    }
  }
  #menu{
    :hover {
      background-color: ${(props) => props.theme.a};
      border-radius: 50%;
    }
  }
  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }
  body {
    margin: 0;
    padding: 0;
    background-color: ${(props) => props.theme.body};
    text-rendering: optimizeLegibility;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
  a {
    text-decoration: none;
    font-weight: bold;
    color: ${(props) => props.theme.a};
  }
  .login {
    color: ${(props) => props.theme.a};
  }
  a:active {
    color: red;
  }
  .nav-menu {
  background-color: ${(props) => props.theme.body};
  width: 200px;
  height: 100vh;
  display: flex;
  position: fixed;
  top: 0;
  left: -100%;
  transition: 850ms;
}

.nav-menu.active {
  left: 0;
  transition: 350ms;
}

`;

export default GlobalStyle;
