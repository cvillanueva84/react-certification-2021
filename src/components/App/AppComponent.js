import React, { lazy, Suspense, useReducer } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import GridLoader from 'react-spinners/ClipLoader';
import { ThemeProvider } from 'styled-components';
import HomePage from '../../pages/Home/HomePage';
import NotFound from '../../pages/NotFound/NotFoundPage';
import AuthProvider from '../../providers/Auth';
import Context from '../../providers/Theme/Theme.context';
import { ThemeReducer } from '../../providers/Theme/Theme.reducer';
import { dark, light } from '../../providers/Theme/themes';
import VideoListProvider from '../../providers/VideoList/VideoList.provider';
import ScrollToTop from '../../utils/scrollToTop';
import FavoriteDetails from '../FavoriteDetails/FavoriteDetailsComponent';
import FavoritesList from '../FavoritesList/FavoritesListComponent';
import Layout from '../Layout';
import Navbar from '../Navbar/NavbarComponent';
import Private from '../Private/PrivateComponent';
import { GlobalStyles, override } from './AppComponent.styled';
const VideoDetails = lazy(() => import('../VideoDetails/VideoDetailsComponent'));

function App() {
  const [state, dispatch] = useReducer(ThemeReducer, {
    isDark: false,
  });

  return (
    <BrowserRouter>
      <AuthProvider>
        <Context.Provider value={{ state, dispatch }}>
          <ThemeProvider theme={state.isDark ? light : dark}>
            <GlobalStyles />
            <VideoListProvider>
              <Layout>
                <Navbar />
                <ScrollToTop />
                <Switch>
                  <Route exact path="/">
                    <HomePage />
                  </Route>
                  <Route exact path="/video/:id">
                    <Suspense fallback={<GridLoader size={150} css={override} />}>
                      <VideoDetails />
                    </Suspense>
                  </Route>
                  <Private exact path="/favorite-videos">
                    <FavoritesList />
                  </Private>
                  <Private exact path="/favorite-video/:id">
                    <FavoriteDetails />
                  </Private>
                  <Route path="*">
                    <NotFound />
                  </Route>
                </Switch>
              </Layout>
            </VideoListProvider>
          </ThemeProvider>
        </Context.Provider>
      </AuthProvider>
    </BrowserRouter>
  );
}

export default App;
