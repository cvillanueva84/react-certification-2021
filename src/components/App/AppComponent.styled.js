import { css } from '@emotion/react';
import { createGlobalStyle } from 'styled-components';

export const override = css`
  display: block;
  margin: 20rem auto;
  border-color: var(--main-blue);
`;

export const GlobalStyles = createGlobalStyle`
  body, #root {
    background: ${({ theme }) => theme.background};
    color: ${({ theme }) => theme.text};
    list-style: none;
  }
  a {
    color: ${({ theme }) => theme.text}
  }

  span {
    color: ${({ theme }) => theme.text}
  }

`;