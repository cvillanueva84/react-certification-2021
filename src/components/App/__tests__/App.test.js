import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';

import App from '../index';

describe('Testing App component', () => {
  let tree;

  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element) => {
      return element;
    });
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });

  beforeEach(() => {
    tree = render(<App />);
  });

  it('should to take snapshop', () => {
    expect(tree).toMatchSnapshot();
  });

  it('should render any JSX', () => {
    expect(tree.getByText('Welcome to the Challenge!').tagName).toBe('H2');
  });
});
