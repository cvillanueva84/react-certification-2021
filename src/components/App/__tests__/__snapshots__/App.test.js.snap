// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component should to take snapshop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-ksluID efXhUb"
      >
        <ul
          class="sc-hBMUJo cnHQZK"
        >
          <li
            class="sc-fnVZcZ eXigRd"
          >
            <button
              class="sc-hKFxyN fZJIyj"
            >
              <i
                class="fas fa-bars fa-2x"
              />
            </button>
          </li>
          <li
            class="sc-fnVZcZ eXigRd"
          >
            <form
              class="sc-bkbkJK yusCV"
            >
              <button
                class="sc-iemWCZ cTDvLi"
                type="submit"
              >
                <i
                  class="fas fa-search fa-lg"
                />
              </button>
              <input
                class="sc-dIvrsQ lbldMX"
                placeholder="Search"
                type="text"
                value="wizeline"
              />
            </form>
          </li>
          <div
            class="sc-fFSPTT inCyZE"
          />
          <div
            class="sc-hHEiqL ezoTYU"
            on="md"
          >
            <li
              class="sc-fnVZcZ eXigRd"
            >
              <div
                class="sc-crzoAE kUFfJh"
              >
                <p
                  class="sc-iqAclL kmNkJM mr-1"
                  header="true"
                >
                  Dark mode
                </p>
                <label
                  class="sc-dIsUp cEcrGb"
                >
                  <input
                    checked=""
                    data-testid="toggle"
                    type="checkbox"
                  />
                  <span
                    class="sc-bqGGPW eQBPTd"
                  />
                </label>
              </div>
            </li>
          </div>
          <div
            class="sc-hHEiqL ezoTYU"
            on="md"
          >
            <li
              class="sc-fnVZcZ eXigRd"
            >
              <button
                class="sc-hKFxyN fZJIyj"
              >
                <i
                  class="fas fa-user-circle fa-2x"
                />
              </button>
            </li>
          </div>
        </ul>
      </nav>
      <div
        class="sc-bdnxRM hCekCk"
      >
        <div>
          <h2
            center="true"
            class="sc-iqAclL kzBsSq title py-1"
            weight="300"
          >
            Welcome to the Challenge!
          </h2>
          <div
            class="sc-eCApnc fuNLsN"
          >
            <div
              class="sc-jSFjdj MLUyO"
            >
              <div
                class="sc-gtsrHT kbMUr"
              >
                <div
                  class="sc-dlnjwi gRGBfx"
                  data-testid="loader"
                />
              </div>
              
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-ksluID efXhUb"
    >
      <ul
        class="sc-hBMUJo cnHQZK"
      >
        <li
          class="sc-fnVZcZ eXigRd"
        >
          <button
            class="sc-hKFxyN fZJIyj"
          >
            <i
              class="fas fa-bars fa-2x"
            />
          </button>
        </li>
        <li
          class="sc-fnVZcZ eXigRd"
        >
          <form
            class="sc-bkbkJK yusCV"
          >
            <button
              class="sc-iemWCZ cTDvLi"
              type="submit"
            >
              <i
                class="fas fa-search fa-lg"
              />
            </button>
            <input
              class="sc-dIvrsQ lbldMX"
              placeholder="Search"
              type="text"
              value="wizeline"
            />
          </form>
        </li>
        <div
          class="sc-fFSPTT inCyZE"
        />
        <div
          class="sc-hHEiqL ezoTYU"
          on="md"
        >
          <li
            class="sc-fnVZcZ eXigRd"
          >
            <div
              class="sc-crzoAE kUFfJh"
            >
              <p
                class="sc-iqAclL kmNkJM mr-1"
                header="true"
              >
                Dark mode
              </p>
              <label
                class="sc-dIsUp cEcrGb"
              >
                <input
                  checked=""
                  data-testid="toggle"
                  type="checkbox"
                />
                <span
                  class="sc-bqGGPW eQBPTd"
                />
              </label>
            </div>
          </li>
        </div>
        <div
          class="sc-hHEiqL ezoTYU"
          on="md"
        >
          <li
            class="sc-fnVZcZ eXigRd"
          >
            <button
              class="sc-hKFxyN fZJIyj"
            >
              <i
                class="fas fa-user-circle fa-2x"
              />
            </button>
          </li>
        </div>
      </ul>
    </nav>
    <div
      class="sc-bdnxRM hCekCk"
    >
      <div>
        <h2
          center="true"
          class="sc-iqAclL kzBsSq title py-1"
          weight="300"
        >
          Welcome to the Challenge!
        </h2>
        <div
          class="sc-eCApnc fuNLsN"
        >
          <div
            class="sc-jSFjdj MLUyO"
          >
            <div
              class="sc-gtsrHT kbMUr"
            >
              <div
                class="sc-dlnjwi gRGBfx"
                data-testid="loader"
              />
            </div>
            
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
